### PR TITLE
swift-api-digester: ignore synthesized type members from conformed clang protocols.

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: test/api-digester/Outputs/clang-module-dump.txt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  "name": "TopLevel",
+  "printedName": "TopLevel",
+  "children": [
+    {
+      "kind": "TypeDecl",
+      "name": "ClangInterface",
+      "printedName": "ClangInterface",
+      "declKind": "Class",
+      "usr": "c:objc(cs)ClangInterface",
+      "location": "\/Volumes\/ssd\/SSDDocument\/sourceroot\/swift\/test\/api-digester\/Inputs\/Foo\/foo.h:7:12",
```
Comment (akyrtzi) [inline test/api-digester/Outputs/clang-module-dump.txt]: These are hardcoded paths from your setup.

File: tools/swift-api-digester/swift-api-digester.cpp [hunk 2]
```diff
@@ -1558,7 +1565,7 @@ static SDKNode *constructTypeAliasNode(SDKContext &Ctx,TypeAliasDecl *TAD) {
 static void addMembersToRoot(SDKContext &Ctx, SDKNode *Root,
                              IterableDeclContext *Context) {
   for (auto *Member : Context->getMembers()) {
-    if (shouldIgnore(Member))
+    if (shouldIgnore(Member, const_cast<Decl*>(Context->getDecl())))
```
Comment (akyrtzi) [inline tools/swift-api-digester/swift-api-digester.cpp]: You don't need to const_cast, just add the the parameter as `static bool shouldIgnore(Decl *D, const Decl* Parent)`

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@akyrtzi I added a new flag to avoid serializing the location of decls.

**Pull request merged**